packages/core/src/unit-of-work/ChangeSetComputer.ts

```diff
@@ -84,7 +84,7 @@ export class ChangeSetComputer {
     const target = changeSet.entity[prop.name] as unknown as Collection<any>;
 
     // remove items from collection based on removeStack
-    if (target.isInitialized()) {
+    if (target.isInitialized() && this.removeStack.size > 0) {
       target.getItems()
```

Comment:
we can improve this bit too - as we already know the collection is initialized, we can skip the init check inside `getItems()`

```suggestion
      target.getItems(false)
```